### ISSUE TITLE
Datapath in debug mode spams "Successfully mapped addr=10.168.0.20 to identity=0" messages

### ISSUE THREAD

**Issue by joestringer** [cilium/cilium#12804]
Version: Cilium v1.8.2

### To reproduce
1. Follow the GKE deployment guide: https://docs.cilium.io/en/v1.8/gettingstarted/k8s-install-gke/
1. Deploy connectivity check and hubble relay via instructions from the above documentation.
1. Edit the ConfigMap to add datapath debug messages:
   ```
      debug: "true"
      debug-verbose: datapath
   ```

### Problem
When I go into one of the nodes and run `cilium monitor -v`, this message floods the output:
```
CPU 02: MARK 0x0 FROM 2462 DEBUG: Successfully mapped addr=10.168.0.20 to identity=0
```

Other messages will show up, but they're drowned out by this message.

In this environment, endpoint `2462` is the host endpoint and the destination is another node's IP address.

**Comment by joestringer** [comment]
```
$ cilium status
KVStore:                Ok   Disabled
Kubernetes:             Ok   1.14+ (v1.14.10-gke.42) [linux/amd64]
Kubernetes APIs:        ["CustomResourceDefinition", "cilium/v2::CiliumClusterwideNetworkPolicy", "cilium/v2::CiliumEndpoint", "cilium/v2::CiliumNetworkPolicy", "cilium/v2::CiliumNode", "core/v1::Endpoint", "core/v1::Namespace", "core/v1::Node", "core/v1::Pods", "core/v1::Service", "networking.k8s.io/v1::NetworkPolicy"]
KubeProxyReplacement:   Probe   [eth0 (DR)]   [NodePort (SNAT, 30000-32767, XDP: DISABLED), HostPort, ExternalIPs, HostReachableServices (TCP, UDP), SessionAffinity]
Cilium:                 Ok      OK
NodeMonitor:            Listening for events on 4 CPUs with 64x4096 of shared memory
Cilium health daemon:   Ok
IPAM:                   IPv4: 11/255 allocated from 10.56.0.0/24,
Masquerading:           BPF   [eth0]   10.56.0.0/14
Controller Status:      62/62 healthy
Proxy Status:           OK, ip 10.56.0.11, 3 redirects active on ports 10000-20000
Hubble:                 Ok              Current/Max Flows: 4096/4096 (100.00%), Flows/s: 29.31   Metrics: Ok
Cluster health:         2/2 reachable   (2020-08-06T17:50:29Z)
```

**Comment by joestringer** [comment]
Perhaps also relevant:

```
$  cilium ip list
IP                 IDENTITY   SOURCE
0.0.0.0/0          2
10.56.0.11/32      1
10.56.0.12/32      4
10.56.0.16/32      28679      k8s
10.56.0.108/32     5759       k8s
10.56.0.119/32     29178      k8s
10.56.0.141/32     8085       k8s
10.56.0.144/32     55632      k8s
10.56.0.152/32     51604      k8s
10.56.0.154/32     7179       k8s
10.56.0.175/32     2643       k8s
10.56.0.219/32     3660       k8s
10.56.1.11/32      50394      k8s
10.56.1.22/32      37076      k8s
10.56.1.26/32      6
10.56.1.70/32      25444      k8s
10.56.1.79/32      27935      k8s
10.56.1.86/32      61293      k8s
10.56.1.127/32     47544      k8s
10.56.1.141/32     42115      k8s
10.56.1.149/32     30857      k8s
10.56.1.173/32     49622      k8s
10.56.1.196/32     27261      k8s
10.56.1.200/32     5759       k8s
10.56.1.224/32     4
10.56.1.226/32     38218      k8s
10.56.1.242/32     42029      k8s
10.168.0.20/32     6
10.168.0.21/32     1
```

**Comment by joestringer** [comment]
It may be that this is the only debug message on the path incoming into the node towards the stack from remote nodes (ie node-to-node), and we don't log any other debug or trace messages. I would expect incoming trace messages in addition to these debug messages, but at least my initial code inspection doesn't reveal where such a trace message would be triggered when hostfw is disabled.

**Comment by joestringer** [comment]
There's another aspect to this which is a bit unusual, specifically the `identity` is determined to be 0 despite the identity being in the ipcache. I'm looking sideways at:

https://github.com/cilium/cilium/blob/3996e5a1a36e60674585173a2cadaa973ab93b9f/bpf/bpf_host.c#L393

@pchaigno do you recall why the `from_host &&` case was added here? My understanding of the other portion is that it's explicitly intended for traffic from remote nodes. Related, the original commit that introduced the check: 5b0689b1204aa

**Comment by joestringer** [comment]
Seems like later on we'll actually look up the ipcache anyway in `ipv4_host_policy_ingress()` and disregard this lookup in the "traffic ingressing from outside the node" case. The logic is also subtly different. This could probably do with a bit more close attention.

**Comment by pchaigno** [comment]
I was able to reproduce with v1.7 and 1.7.0. The message is a bit different in v1.7.0; `Failed to map` instead of `Successfully mapped`. It is also not required to deploy the connectivity check to reproduce. As far as I understand, the messages corresponded to the SSH packets in my case, thus the high rate.

<br>

> @pchaigno do you recall why the `from_host &&` case was added here? My understanding of the other portion is that it's explicitly intended for traffic from remote nodes. Related, the original commit that introduced the check: 5b0689b

If was added because of the `#ifndef ENABLE_EXTRA_HOST_DEV` one line guard above. From the commit that introduced that `from_host` (a695f532d0607):

> `ENABLE_EXTRA_HOST_DEV` was only defined in `init.sh` for the `from_host` path.
It is now defined from the Go side for all paths, and we check whether
`from_host` is true in the C code instead.

<br>

> Seems like later on we'll actually look up the ipcache anyway in ipv4_host_policy_ingress() and disregard this lookup in the "traffic ingressing from outside the node" case.

Yep. I still have an item on my TODO to rework this and avoid the unnecessary lookup (though it predates the host endpoint IIRC), but like you said, it requires close attention.

Given we report a 0 identity, this rework should allow us to also avoid the excessive debug messages.
